src/backend/TrafficCourts/Citizen.Service/Controllers/DisputesController.cs

```diff
@@ -271,11 +271,33 @@ public async Task<IActionResult> GetDisputeAsync(string guidHash, CancellationTo
                 return NotFound("Dispute not found");
             }
 
-            //var givenNames = response.Message.DisputantGivenName1 + " " + response.Message.DisputantGivenName2 + " " + response.Message.DisputantGivenName3;
-            //if (response.Message.DisputantSurname != user?.Surename
-            //    || !(response.Message.DisputantGivenName1 == user?.GivenName || givenNames.TrimEnd() == user?.GivenNames))
+            // TODO:  Uncomment before UAT and go-live
+            //// if contact type is individual then match with disputant name otherwise match with contact names
+            //if (response.Message.ContactTypeCd == DisputeContactTypeCd.INDIVIDUAL)
             //{
-            //    return BadRequest("Disputant not match");
+            //    var givenNames = response.Message.DisputantGivenName1
+            //        + (response.Message.DisputantGivenName2 != null ? (" " + response.Message.DisputantGivenName2) : "")
+            //        + (response.Message.DisputantGivenName3 != null ? (" " + response.Message.DisputantGivenName3) : "");
+            //    if (response.Message.DisputantSurname.ToUpper() == user?.Surname.ToUpper()

```

Comment:
Do not use the convert to common case comparison pattern. Prefer using case insensitive comparisons, such as:

```cs
    bool eq;
    string a = "a";
    string b = "b";
    // using StringComparer
    eq = StringComparer.OrdinalIgnoreCase.Equals(a, b);
    // or StringComparison
    eq = a.Equals(b, StringComparison.OrdinalIgnoreCase);
```

you can  simplify this by creating a local variable for the StringComparer, `var comparer = StringComparer.OrdinalIgnoreCase` and then using that, `comparer.Equals(a, b)`

In addition, you could encapsulate this logic into an extension method, like

```cs
public static class DisputantExtensions
{
   ///<summary>Checks to see if the message dispuant name matches the user's username</summary>
   public static bool SameName(this WhatEverTheTypeOfMessageIs message, UserType user)
   {
       // logic
   }
}
```

This makes it easier to create a unit test to validate the name comparison is working as expected instead of having to deal with that in the controller test.